src/test/java/com/worksap/fig/SeqTest.java

```diff
@@ -117,6 +124,37 @@ public void testContainsAny() {
     }
 
     @Test
+    public void testContainSubSeq() {
+        assertTrue(Seqs.newSeq('B', 'B', 'C', ' ', 'A', 'B', 'C', 'D', 'A', 'B', ' ', 'A', 'B', 'C', 'D', 'A', 'B', 'C', 'D', 'A', 'B', 'D', 'E')
+                .containsSubSeq(Seqs.newSeq('A', 'B', 'C', 'D', 'A', 'B', 'D')));
+        assertFalse(Seqs.newSeq('B', 'B', 'C', ' ', 'A', 'B', 'C', 'D', 'A', 'B', ' ', 'A', 'B', 'C', 'D', 'A', 'B', 'C', 'D', 'A', 'B', 'D', 'E')
+                .containsSubSeq(Seqs.newSeq('A', 'B', 'C', 'D', 'A', 'B', 'A')));
+        assertFalse(Seqs.newSeq('A')
+                .containsSubSeq(Seqs.newSeq('B')));
+        assertFalse(Seqs.newSeq()
+                .containsSubSeq(Seqs.newSeq('B')));
+        assertTrue(Seqs.newSeq('A')
+                .containsSubSeq(Seqs.newSeq()));
+        assertTrue(Seqs.newSeq('A', 'B', 'C')
+                .containsSubSeq(Seqs.newSeq('A', 'B', 'C')));
+
+        assertThrows(NullPointerException.class, () -> Seqs.newSeq('A', 'B', 'C').containsSubSeq(null));
+
+        assertEquals(15, Seqs.newSeq('B', 'B', 'C', ' ', 'A', 'B', 'C', 'D', 'A', 'B', ' ', 'A', 'B', 'C', 'D', 'A', 'B', 'C', 'D', 'A', 'B', 'D', 'E')
+                .indexOfSubSeq(Seqs.newSeq('A', 'B', 'C', 'D', 'A', 'B', 'D')));
+        assertEquals(15, Seqs.newSeq('B', 'B', 'C', ' ', 'A', 'B', 'C', 'D', 'A', 'B', ' ', 'A', 'B', 'C', 'D', 'A', 'B', 'C', 'D', 'A', 'B', 'D', 'E')
+                .lastIndexOfSubSeq(Seqs.newSeq('A', 'B', 'C', 'D', 'A', 'B', 'D')));
+
+        assertEquals(1, Seqs.newSeq(3, 1, 2, 1, 2, 1, 2, 3)
```

Comment:
No case of empty and larger pattern seq. For unit test, we can consider hamcrest that is used by junit 4.